src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -395,6 +395,12 @@ private async Task ProduceEndAwaited()
         await _bodyOutput.FlushAsync(default);
     }
 
+    // Response trailers, reset, and GOAWAY are only on HTTP/2+ and require IIS support
+    // that is only available on Win 11/Server 2022 or later.
+    protected bool AdvancedHttp2FeaturesSupported => OperatingSystem.IsWindowsVersionAtLeast(10, 0, 20348, 0) &&
```

Comment:
Could cache the version check in a static as it's presumably not going to change at runtime?